link/kprobe.go

```diff
@@ -144,6 +138,34 @@ func Kretprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions) (Link, er
 	return lnk, nil
 }
 
+func isValidSymbol(s string) bool {
+	// The code below produces the same results as
+	// regexp.MustCompile("^[a-zA-Z_][0-9a-zA-Z_.]*$").MatchString(s).
+
+	if len(s) < 1 {
+		return false
+	}
+
+	for i := 0; i < len(s); i++ {
```

Comment:
Nit: `for i, c := range []byte{s} {`

Casting to []byte in a range loop doesn't allocate.

Reply:
You meant ` range []byte(s)` I guess. Yes, this looks much better!